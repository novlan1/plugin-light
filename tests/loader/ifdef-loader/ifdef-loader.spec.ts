import path from 'path';
import preprocessLoader from '../../../src/webpack-loader/ifdef-loader';

jest.mock('loader-utils', () => ({
  getOptions: jest.fn()
    .mockReturnValueOnce({
      type: ['js'],
      num: 1,
      context: { H5: true },
    })
    .mockReturnValueOnce({
      context: { H5: true },
      num: 2,
    })
    .mockReturnValueOnce({
      type: ['js'],
      context: { 'MP-WEIXIN': true },
      num: 3,
    })
    .mockReturnValueOnce({
      type: ['html'],
      context: { H5: true },
      num: 4,
    })
    .mockReturnValueOnce({
      type: ['html'],
      context: { 'MP-WEIXIN': true },
      num: 5,
    })
    .mockReturnValueOnce({
      type: ['js'],
      context: { 'MP-WEIXIN': true },
      num: 6,
    })
    .mockReturnValueOnce({
      type: ['css'],
      context: { H5: true },
      num: 7,
    })
    .mockReturnValueOnce({
      type: ['css'],
      context: { 'MP-WEIXIN': true },
      num: 8,
    })
    .mockReturnValueOnce({
      type: ['html'],
      context: { H5: true },
      num: 9,
    })
    .mockReturnValueOnce({
      type: ['html', 'css', 'js'],
      context: { H5: true },
      num: 10,
    })
    .mockReturnValueOnce({
      type: ['html', 'css', 'js'],
      context: { 'MP-WEIXIN': true },
      num: 11,
    })
    .mockReturnValue({
      type: ['js'],
      context: { 'MP-WEIXIN': true },
    }),
}));


const jsStr = `
// #ifdef H5
console.log('1');
// #endif

// #ifndef H5
console.log('2');
// #endif

console.log('3');
`;


const htmlStr = `
<!-- #ifdef H5 -->
console.log('a');
<!-- #endif -->


<!-- #ifndef H5 -->
console.log('b');
<!-- #endif -->

console.log('c');
`;


const cssStr = `
/* #ifdef H5 */
console.log('a.1');
/* #endif */

/* #ifndef H5 */
console.log('b.1');
/* #endif */

console.log('c.1');
`;

const vueStr = `
${jsStr}

${htmlStr}

${cssStr}
`;


describe('preprocessLoader.js', () => {
  it('preprocessLoader.1.jsStr', () => {
    expect(preprocessLoader.call({
      resourcePath: path.resolve(process.cwd(), './src', 'MOCK_PAGE.vue'),
    }, jsStr)).toMatchSnapshot();
  });

  it('preprocessLoader.2.jsStr', () => {
    expect(preprocessLoader.call({
      resourcePath: path.resolve(process.cwd(), './src', 'MOCK_PAGE.vue'),
    }, jsStr)).toMatchSnapshot();
  });

  it('preprocessLoader.3.jsStr', () => {
    expect(preprocessLoader.call({
      resourcePath: path.resolve(process.cwd(), './src', 'MOCK_PAGE.vue'),
    }, jsStr)).toMatchSnapshot();
  });

  it('preprocessLoader.4.htmlStr', () => {
    expect(preprocessLoader.call({
      resourcePath: path.resolve(process.cwd(), './src', 'MOCK_PAGE.vue'),
    }, htmlStr)).toMatchSnapshot();
  });

  it('preprocessLoader.5.htmlStr', () => {
    expect(preprocessLoader.call({
      resourcePath: path.resolve(process.cwd(), './src', 'MOCK_PAGE.vue'),
    }, htmlStr)).toMatchSnapshot();
  });

  it('preprocessLoader.6.htmlStr', () => {
    expect(preprocessLoader.call({
      resourcePath: path.resolve(process.cwd(), './src', 'MOCK_PAGE.vue'),
    }, htmlStr)).toMatchSnapshot();
  });
  it('preprocessLoader.7.cssStr', () => {
    expect(preprocessLoader.call({
      resourcePath: path.resolve(process.cwd(), './src', 'MOCK_PAGE.vue'),
    }, cssStr)).toMatchSnapshot();
  });
  it('preprocessLoader.8.cssStr', () => {
    expect(preprocessLoader.call({
      resourcePath: path.resolve(process.cwd(), './src', 'MOCK_PAGE.vue'),
    }, cssStr)).toMatchSnapshot();
  });
  it('preprocessLoader.9.cssStr', () => {
    expect(preprocessLoader.call({
      resourcePath: path.resolve(process.cwd(), './src', 'MOCK_PAGE.vue'),
    }, cssStr)).toMatchSnapshot();
  });
  it('preprocessLoader.10.vueStr', () => {
    expect(preprocessLoader.call({
      resourcePath: path.resolve(process.cwd(), './src', 'MOCK_PAGE.vue'),
    }, vueStr)).toMatchSnapshot();
  });
  it('preprocessLoader.11.vueStr', () => {
    expect(preprocessLoader.call({
      resourcePath: path.resolve(process.cwd(), './src', 'MOCK_PAGE.vue'),
    }, vueStr)).toMatchSnapshot();
  });
});

