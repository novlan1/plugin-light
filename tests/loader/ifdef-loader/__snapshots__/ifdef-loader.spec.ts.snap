// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`preprocessLoader.js preprocessLoader.1.jsStr 1`] = `
"

console.log('1');






console.log('3');
"
`;

exports[`preprocessLoader.js preprocessLoader.2.jsStr 1`] = `
"
// #ifdef H5
console.log('1');
// #endif

// #ifndef H5
console.log('2');
// #endif

console.log('3');
"
`;

exports[`preprocessLoader.js preprocessLoader.3.jsStr 1`] = `
"





console.log('2');


console.log('3');
"
`;

exports[`preprocessLoader.js preprocessLoader.4.htmlStr 1`] = `
"

console.log('a');







console.log('c');
"
`;

exports[`preprocessLoader.js preprocessLoader.5.htmlStr 1`] = `
"






console.log('b');


console.log('c');
"
`;

exports[`preprocessLoader.js preprocessLoader.6.htmlStr 1`] = `
"
<!-- #ifdef H5 -->
console.log('a');
<!-- #endif -->


<!-- #ifndef H5 -->
console.log('b');
<!-- #endif -->

console.log('c');
"
`;

exports[`preprocessLoader.js preprocessLoader.7.cssStr 1`] = `
"

console.log('a.1');






console.log('c.1');
"
`;

exports[`preprocessLoader.js preprocessLoader.8.cssStr 1`] = `
"





console.log('b.1');


console.log('c.1');
"
`;

exports[`preprocessLoader.js preprocessLoader.9.cssStr 1`] = `
"
/* #ifdef H5 */
console.log('a.1');
/* #endif */

/* #ifndef H5 */
console.log('b.1');
/* #endif */

console.log('c.1');
"
`;

exports[`preprocessLoader.js preprocessLoader.10.vueStr 1`] = `
"


console.log('1');






console.log('3');




console.log('a');







console.log('c');




console.log('a.1');






console.log('c.1');

"
`;

exports[`preprocessLoader.js preprocessLoader.11.vueStr 1`] = `
"






console.log('2');


console.log('3');









console.log('b');


console.log('c');








console.log('b.1');


console.log('c.1');

"
`;
